internal/vds/subvolume.cpp

```diff
@@ -13,13 +13,13 @@ int lineno_annotation_to_voxel(
     Axis const& axis
 ) {
     /* Assume that annotation coordinates are integers */
-    int min      = axis.min();
-    int max      = axis.max();
+    float min    = axis.min();
+    float max    = axis.max();
     int nsamples = axis.nsamples();
+    auto stride  = (max - min) / (nsamples - 1);
+    float index  = (lineno - min) / stride;
 
-    auto stride = (max - min) / (nsamples - 1);
-
-    if (lineno < min || lineno > max || (lineno - min) % stride) {
+    if (lineno < min || lineno > max || std::floor(index) != index) {
         throw std::runtime_error(
             "Invalid lineno: " + std::to_string(lineno) +
```

Comment:
Why do we even test this in _e2e_ tests? This should be a unit test.

I don't know how common it is, but silently floor it to an int is wrong so we need to deal with floats

Reply:
Because you got unlucky.
In E2E we just test what happens in various paths when some error happens. That error was chosen as "some".

Though `main_test.go` and e2e tests overlapped much originally. Now as we extracted setup to `setupApp` they seem to be repeating themselves even more so. Would need to rethink if it makes sense to try to reduce this overlap more.
Go tests are cool because they give you answer fast, but proper E2E are the backup security and actually work with Azure instead of local files, so you don't want to murder them completely.

As to where this error message appears. Functionality happens to be tested 3 times:
- In proper E2E test, full message is tested as a chosen example of some error. Fails.
- In main_test.go also as an error example. But there "index" is provided, not "annotation", so no change.
- In vds_test.go all possible unit tests happen. The full message is never tested.

So we discovered something that was intentionally never unit-tested explicitly before it was merged. So I think this side-effect is not exactly a bad thing.

I think we in error message can just format it with 2 characters after comma and it would look nicer.